internal/gcsx/temp_file.go

```diff
@@ -71,6 +74,14 @@ type StatResult struct {
 	Mtime *time.Time
 }
 
+// Metadata store struct
+type Metadata struct {
```

Comment:
Metadata is a broad and ambiguous term, especially in the package "gcsx". It can be the metadata of the object, the temp file, or something else. In this case, we are creating this struct for ObjectMetadata used by TempFile. Therefore, we can fix this in any of the three ways:

(1) Use "TempFileObjectMetadata"
(2) Move the temp_file.go into a new package "tempfile", and keep the name "Metadata" - "tempfile.ObjectMetadata" also clear
(3) Keep it private by changing the name to "objectMetadata".